app/uk/gov/hmrc/channelpreferences/controllers/PreferenceController.scala

```diff
@@ -84,4 +91,39 @@ class PreferenceController @Inject()(
         }
     }
   }
+
+  def update(key: String, status: String): Action[AnyContent] =
+    Action.async { request =>
+      val correlationId = extractCorrelationId(request)
+      key.toUpperCase() match {
+        case ITSA_REGIME => handleItsaStatus(status, correlationId)
+        case _           => Future.successful(BadRequest(s"The key $key is not supported"))
+      }
+    }
+
+  private def extractCorrelationId(request: Request[AnyContent]): String = {
+    val ACKNOWLEDGEMENT_REFERENCE_MAX_LENGTH = 32
+    val randomId = UUID.randomUUID().toString
+    request.headers
+      .get(CustomHeaders.CorrelationId)
+      .getOrElse(randomId)
+      .replace("-", "")
+      .substring(0, ACKNOWLEDGEMENT_REFERENCE_MAX_LENGTH - 1)
+  }
+
+  private def eisUpdateContact(status: Boolean, correlationId: String): Future[Result] =
+    eisConnector.updateContactPreference(ITSA_REGIME.toLowerCase, status, correlationId).map {
+      case Right(_) => Ok
+      case Left(EisUpdateContactError(message)) =>
+        logger.warn(message)
```

Comment:
could we raise an error log instead of a warn level one?

Reply:
Done in https://github.com/hmrc/channel-preferences/pull/34/commits/e704e16203d192f2f8222a02d6a1450ccfd79caf